plugins/events/rest.go

```diff
@@ -122,3 +125,72 @@ func eventsListHandler(w http.ResponseWriter, r *http.Request) {
 
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, events)
 }
+
+func checkConnection(c transaction.TxnCtx) error {
+	var req eventsapi.Webhook
+
+	if err := c.Get("req", &req); err != nil {
+		return err
+	}
+	e := gd2events.New("TEST", map[string]string{}, false)
+	return gd2events.WebhookPublish(&req, e)
+}
+
+func registerWebhookTestStepFuncs() {
+	var sfs = []struct {
+		name string
+		sf   transaction.StepFunc
+	}{
+		{"webhook-test.checkConnection", checkConnection},
+	}
+	for _, sf := range sfs {
+		transaction.RegisterStepFunc(sf.sf, sf.name)
+	}
+}
+
+func webhookTestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	var req eventsapi.Webhook
+
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(
+			ctx, w, http.StatusBadRequest,
+			errors.ErrJSONParsingFailed)
+		return
+	}
+
+	if req.URL == "" {
```

Comment:
TRIVIAL: Something I read recently, may be usefull, instead of comparing string like this its faster to compare the length of string, though internally this comparison happens (implicit). Also we can trim the strings before comparing,

Reply:
I just used the code from [here](https://github.com/gluster/glusterd2/blob/master/plugins/events/rest.go#L27 ). Its a matter of taste,  both will generate the same code [check  here](https://groups.google.com/forum/#!topic/golang-nuts/5bxsM_sPQyI)